### DISCUSSION TITLE
Mopup vs Maui Community Popup

### DISCUSSION THREAD

**Discussion by gktval** [LuckyDucko/Mopups#52] [General]
Love the Mopup. I wish I had found it sooner. There are so many little problems with the Maui Community Toolkit Popup, such as layout stuff, showing multiple popups, and crashes. 

One things I did learn to love from the Maui Community Toolkit Popup, that might be nice to see in Mopup is the Sync Push. For instance, if I want to await a result from the popup. In the Popup class there is an Object Result property that is passed when the user awaits for the dialog to close. I was able to inherit from Popup Navigation to do the same thing. Below is my MopupNavigation class that returns a bool Result. Otherwise, awesome project!

```
public class MopupNavigation : PopupNavigation
{

    private TaskCompletionSource<bool> _taskSource = new TaskCompletionSource<bool>();
    public async Task<bool> PushAsync(Popup popup)
    {
        popup.Disappearing += (s, e) => OnDisappear(popup);
        await this.PushAsync(popup, true);
        return await _taskSource.Task;
    }

    public async void Push(Popup popup)
    {
        popup.Disappearing += (s, e) => OnDisappear(popup);

        await this.PushAsync(popup, true);
    }


    private void OnDisappear(Popup popup)
    {
        if (!_taskSource.Task.IsCompleted)
            _taskSource.SetResult(popup.Result);
    }

    internal void Pop(Popup popup, bool result)
    {
        popup.Result = result;
        this.PopAsync();
    }
}
```

**Comment by LuckyDucko** [comment]
Here is the [discussion](https://github.com/LuckyDucko/Mopups/discussions/8)  we are having about this 
Previously I created the [AwaitablePopups](https://github.com/LuckyDucko/AwaitablePopups), however it adds a bit of overhead in requirements with the forced need for a specific ViewModel and whatnot.

I definitely see the need for having values returned from popups, jury is out whether it should be built in or supplied by others
